components/script/dom/htmlselectelement.rs

```diff
@@ -183,6 +184,51 @@ impl HTMLSelectElementMethods for HTMLSelectElement {
     fn Labels(&self) -> Root<NodeList> {
         self.upcast::<HTMLElement>().labels()
     }
+
+    // https://html.spec.whatwg.org/multipage/#dom-select-length
+    fn SetLength(&self, value: u32) {
+        let length  = self.Length();
+        let node = self.upcast::<Node>();
+        if value < length {     // truncate the number of option elements
+            for _ in 0..(length - value) {
+                if let Some(lastChild) = node.GetLastChild() {
```

Comment:
I didn't ignore your comment, check my reply to your previous comment. It doesn't work as you mentioned children nodes aren't deleted.